internal/servicelog/ringbuffer.go

```diff
@@ -342,7 +342,12 @@ func (rb *RingBuffer) releaseIterators() {
 	rb.iteratorMutex.Lock()
 	defer rb.iteratorMutex.Unlock()
 	for _, iter := range rb.iteratorList {
-		close(iter.closeChan)
+		// Close closeChan if not already closed
```

Comment:
True. This change was because after you close the ring buffer for writes (`RingBuffer.Close`), you should still be able to create read iterators, and this wasn't working due to the double close, hence this fix.

However, I've discussed with @hpidcock, and there is a better way to structure the RingBuffer to avoid the double close, as well as combine the individual `Iterator.closeChan` into one (semantically it should probably be one channel on the ring buffer). This change isn't hurting, so I'll leave it here, but we plan to refactor that in a subsequent commit.